packages/api/lambdas/index-from-database.js

```diff
@@ -25,33 +40,113 @@ async function indexModel(esClient, tableName, esIndex, indexFn) {
       scanQueue.items.pop();
     }
 
-    log.info(`Indexing ${scanQueue.items.length} records from ${tableName}`);
-
-    await Promise.all(scanQueue.items.map((item) => indexFn(esClient, item, esIndex)));
+    if (scanQueue.items.length === 0) {
+      log.info(`No records to index for ${tableName}`);
+      return true;
+    }
 
-    log.info(`Completed index of ${scanQueue.items.length} records from ${tableName}`);
+    log.info(`Attempting to index ${scanQueue.items.length} records from ${tableName}`);
+
+    const input = scanQueue.items.map(
+      (item) => limitEsRequests(
+        async () => {
+          try {
+            return await indexFn(esClient, item, esIndex);
+          } catch (err) {
+            log.error(`Error indexing record ${JSON.stringify(item)}, error: ${err}`);
+            return false;
+          }
+        }
+      )
+    );
+    const results = await Promise.all(input);
+    const successfulResults = results.filter((result) => result !== false);
+    totalItemsIndexed += successfulResults;
+
+    log.info(`Completed index of ${successfulResults.length} records from ${tableName}`);
   }
   /* eslint-enable no-await-in-loop */
+
+  return totalItemsIndexed;
 }
 
-async function indexFromDatabase(esIndex, tables, esHost) {
+async function indexFromDatabase({
+  esIndex,
+  tables,
+  esHost,
+  esRequestConcurrency = 10
+}) {
   const esClient = await Search.es(esHost);
+  const limitEsRequests = pLimit(esRequestConcurrency);
 
   await Promise.all([
-    indexModel(esClient, tables.collectionsTable, esIndex, indexer.indexCollection),
-    indexModel(esClient, tables.executionsTable, esIndex, indexer.indexExecution),
-    indexModel(esClient, tables.asyncOperationsTable, esIndex, indexer.indexAsyncOperation),
-    indexModel(esClient, tables.granulesTable, esIndex, indexer.indexGranule),
-    indexModel(esClient, tables.pdrsTable, esIndex, indexer.indexPdr),
-    indexModel(esClient, tables.providersTable, esIndex, indexer.indexProvider),
-    indexModel(esClient, tables.rulesTable, esIndex, indexer.indexRule)
+    indexModel({
+      esClient,
+      tableName: tables.collectionsTable,
+      esIndex,
+      indexFn: indexer.indexCollection,
+      limitEsRequests
```

Comment:
I really dislike passing the `p-limit` function as a parameter, but it seemed wasteful to pass the limit value into `indexModel` and re-initialize the concurrency function every time